### PR TITLE
fix: round robin re assignment google meet bug

### LINKED ISSUE #17377: [CAL-4636] Booking manual re assignment with google meet
Create a round robin event type and set location to organizer's default application and any two users as round robin hosts.

Connect google calendar and google meet for two round robin hosts and set google meet to default conferencing tool.

Now book a meeting and then re assign it to another person.

Now check the email. it would contain the cal video url instead of google meet and on booking success page you would see old google meet link.

[https://calcominc.slack.com/archives/C07SC43QSKU/p1729867569648709](https://calcominc.slack.com/archives/C07SC43QSKU/p1729867569648709)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4636](https://linear.app/calcom/issue/CAL-4636/booking-re-assignment-with-google-meet)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

- Fixes  https://github.com/calcom/cal.com/issues/17377 (GitHub issue number)
- Fixes CAL-4636 (Linear issue number - should be visible at the bottom of the GitHub issue description)

BEFORE:- Adds cal video

Screenshot 2024-10-29 at 1 08 22 AM

Screenshot 2024-10-29 at 12 44 32 AM

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- mentioned in the issue description

Check all email's body

### CODE REVIEW

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts [hunk 1]
```diff
@@ -134,8 +138,7 @@ export const roundRobinManualReassignment = async ({
     const responseSafeParse = await responseSchema.safeParseAsync(bookingResponses);
     const responses = responseSafeParse.success ? responseSafeParse.data : undefined;
 
-    let bookingLocation = booking.location;
-    if (eventType.locations.includes({ type: OrganizerDefaultConferencingAppType })) {
```
Comment (Udit-takkar) [inline packages/features/ee/round-robin/roundRobinManualReassignment.ts:138]: includes compare object by reference and not value so this won't ever be true even when it contains an object `{ type: OrganizerDefaultConferencingAppType }`

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts [hunk 2]
```diff
@@ -262,22 +266,111 @@ export const roundRobinManualReassignment = async ({
...
+  if (results.length) {
+    // Handle Google Meet results
+    // We use the original booking location since the evt location changes to daily
+    if (bookingLocation === MeetLocationType) {
+      const googleMeetResult = {
+        appName: GoogleMeetMetadata.name,
+        type: "conferencing",
+        uid: results[0].uid,
+        originalEvent: results[0].originalEvent,
+      };
```
Comment (Udit-takkar) [inline packages/features/ee/round-robin/roundRobinManualReassignment.ts:295]: Using the same code from handleNewBooking

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts [hunk 3]
```diff
@@ -262,22 +266,111 @@ export const roundRobinManualReassignment = async ({
...
     bookingId,
     newReferencesToCreate,
   });
 
-  const { cancellationReason, ...evtWithoutCancellationReason } = evt;
+  prisma.booking.update({
+    where: { id: bookingId },
+    data: { metadata },
+  });
+
```
Comment (Udit-takkar) [inline packages/features/ee/round-robin/roundRobinManualReassignment.ts:367]: update the new metadata to make sure new google meet url is saved

**Review by Udit-takkar — COMMENTED** [review]
Self review done

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts (hunk 2 above)
Comment (keithwillcode) [reply packages/features/ee/round-robin/roundRobinManualReassignment.ts:295]: Can we reuse this code instead of copy/pasting?

Comment (keithwillcode) [reply packages/features/ee/round-robin/roundRobinManualReassignment.ts:295]: This is important logic that ideally lives in 1 place

Comment (Udit-takkar) [reply packages/features/ee/round-robin/roundRobinManualReassignment.ts:295]: handleNewBooking is also doing a lot of things during rescheduling.  It's mutating a lot of variables which could be difficult to track.

**Review by joeauyeung — DISMISSED** [review]
I'm reading through the code here and in `handleNewBooking` and the problem is that we're applying additional logic to the return array of `referencesToCreate` from the `EventManager`. Could we instead move this logic inside of the `EventManager` that way we don't have multiple instances of this logic?

**Review by CarinaWolli — DISMISSED** [review]
- I still got the wrong conferencing app when the two host's had different default apps

- When I try to reassign my booking (both google meet default conferencing app), I am getting an error:
Screenshot 2024-10-29 at 3 12 03 PM

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts [hunk 4]
```diff
@@ -248,36 +247,34 @@ export const roundRobinManualReassignment = async ({
...
 
-  const eventManager = new EventManager({ ...newUser, credentials });
   const previousHostDestinationCalendar = hasOrganizerChanged
     ? await prisma.destinationCalendar.findFirst({
         where: { userId: originalOrganizer.id },
       })
     : null;
 
-  const results = await eventManager.reschedule(
+  const { evtWithAdditionalInfo } = await handleRescheduleEventManager({
```
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply packages/features/ee/round-robin/roundRobinManualReassignment.ts:264]: yeah added

**Comment by Udit-takkar** [comment]
@CarinaWolli  everything should be fixed now. i was passing wrong metadata while updating booking

**Review by joeauyeung — APPROVED** [review]
@Udit-takkar @CarinaWolli I tested this again and it's working as expected. @CarinaWolli has pointed out this doesn't work if the two users have different conferencing apps but for now this fixes the issue when two users have Google Meet set as their default conferencing app.

**Pull request merged**